### PR TITLE
Logseq JSON Importer

### PR DESCRIPTION
**Pull request by tekacs**
Based substantially on the work of @cheslip and @sawhney17. :)

Should resolve most issues associated with Logseq import, 'allow edits by maintainers' is turned on.

There are a few things that could be tweaked further, happy to jump in on them as they come up. In particular I wonder about how much should be factored out in common and how implicit the handling should be of divergent behaviour in the common code between the Roam and Logseq importers. Also, file layout around `src/converters/common.ts`.

Will tidy up this PR message later!

### CODE REVIEW

**Comment by sawhney17** [comment]
Hmm, I'm unable to run the export at all, it's' returning a blank file. 
Screenshot 2022-10-26 at 11 19 13 AM

**Comment by sawhney17** [comment]
Never mind, ran the Roam command 😅

**Comment by eirikhm** [comment]
Hey @tekacs! Sorry for the late reply, it's been crazy! I'll try to do a review of this either today or early next week. If you have any questions, just ping me on Slack and I'll do my best to assist!

File: src/converters/common.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  'September',
+  'October',
+  'November',
+  'December',
+];
+
+const monthPrefixes = months.map((m) => m.slice(0, 3))
+
+// Convert 'June 1st, 2021' and 'Jun 1st, 2021' to 'MM-DD-YYYY' without dealing with timezones, etc.
+export function dateStringToRoamDateUID(str: string) {
```
Comment (eirikhm) [inline src/converters/common.ts:27]: This is no longer Roam-specific, so perhaps we should find a better name?

File: src/converters/logseq/logseqUtils.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+export function setNodeAsTodo(node: TanaIntermediateNode) {
+  node.name = node.name.substring(TODO_FLAG.length);
+  node.todoState = 'todo';
+}
+export function setNodeAsDone(node: TanaIntermediateNode) {
+  node.name = node.name.substring(DONE_FLAG.length);
+  node.todoState = 'done';
+}
+
+export function replaceRoamSyntax(nameToUse: string) {
```
Comment (eirikhm) [inline src/converters/logseq/logseqUtils.ts:29]: Also no longer specific to the Roam converter

File: src/converters/logseq/index.ts [hunk 3]
```diff
@@ -0,0 +1,629 @@
+import { exit } from 'process';
```
Comment (eirikhm) [inline src/converters/logseq/index.ts:1]: This class uses terms like `roamToIntermediate` which will probably be confusing for people. Would be great if we can get those renamed. 

It also has more bits from the roam converter, like `extractMetaNodeContentAndGetNumRemaningChildren` and dealing with `#`. If these features are present in logseq as well we should probably split out the code or add tests that verify them here.

**Comment by tekacs** [comment]
I missed that this review had happened until today @eirikhm -- I'll take a look at resolving this out soon. :)

**Comment by jesper-bylund** [comment]
@tekacs First, thank you for the great work. :) Used it with good results! Are you still looking at this review?

**Comment by jesper-bylund** [comment]
Sorry @eirikhm, making a mess of these naming tweaks. Unused to GitHub spaces. Will make a separate PR based on this one.

**Pull request merged**